### PR TITLE
chore(fiscal): polish user interface

### PR DESCRIPTION
**Pull request by jniles**
This commit polishes the user interface and updates the code to the
latest ESLint checks in the client.  The most significant changes are:
 1. Title accounts are highlighted on the opening/closing balance grids.
 2. Performance increases on the UI-grids via template optimisations and
 fastWatch.
 3. The "Close Fiscal Year" button is hidden if a Fiscal Year is closed.
 4. Removes aggregation labels from footers.

Partially addresses #2490.  References #715.

### CODE REVIEW

File: client/src/modules/fiscal/fiscal.closingBalance.js [hunk 1]
```diff
@@ -84,11 +85,15 @@ function FiscalClosingBalanceController($state, Accounts, AccountStore, Fiscal,
...
     })
-    .then(function (list) {
-      vm.accounts = list;
-      vm.gridOptions.data = Accounts.order(vm.accounts);
-    })
-    .catch(Notify.handleError);
+      .then((list) => {
+        list.forEach(account => {
+          account.isTitleAccount = account.type_id === bhConstants.accounts.TITLE;
+        });
```
Comment (mbayopanda) [inline client/src/modules/fiscal/fiscal.closingBalance.js:91]: 👍

File: client/src/modules/fiscal/fiscal.manage.html [hunk 2]
```diff
@@ -41,7 +41,7 @@
             <select class="form-control"
               ng-model="FiscalManageCtrl.fiscal.previous_fiscal_year_id"
               ng-options="fy.id as fy.hrLabel for fy in FiscalManageCtrl.previous_fiscal_year">
-              <option value="" disabled>--{{ 'FORM.SELECT.FISCAL_YEAR' | translate }}--</option>
+              <option value="" disabled translate>FORM.SELECT.FISCAL_YEAR</option>
```
Comment (mbayopanda) [inline client/src/modules/fiscal/fiscal.manage.html:44]: 👍

File: client/src/modules/fiscal/fiscal.manage.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <form name="FiscalForm" bh-submit="FiscalManageCtrl.submit(FiscalForm)" novalidate>
 
   <div class="row">
-    <div class="col-md-12">
+    <div class=" col-lg-6 col-md-10 col-sm-12 col-xs-12">
```
Comment (mbayopanda) [inline client/src/modules/fiscal/fiscal.manage.html:4]: Good 👍

File: client/src/modules/fiscal/fiscal.manage.html [hunk 4]
```diff
@@ -66,7 +66,7 @@
       <p><strong translate>FORM.INFO.OTHER_ACTIONS</strong></p>
       <p>
         <!-- close the fiscal year  -->
-        <a class="btn btn-danger" data-action="closing-fiscal-year" href ng-click="FiscalManageCtrl.closingFiscalYear()">
+        <a class="btn btn-danger" data-action="closing-fiscal-year" href ng-click="FiscalManageCtrl.closingFiscalYear()" ng-hide="FiscalManageCtrl.fiscal.locked">
```
Comment (mbayopanda) [inline client/src/modules/fiscal/fiscal.manage.html:69]: Good catch

File: client/src/modules/fiscal/fiscal.manage.js [hunk 5]
```diff
@@ -6,24 +6,22 @@ FiscalManagementController.$inject = [
...
+ * provides utility functions for submission and error handling.
  */
 function FiscalManagementController($state, Fiscal, Notify, Modal, util, moment) {
-  var vm = this;
-  var id;
-  var isUpdate;
+  const vm = this;
 
-  // state variables
+  // state constiables
```
Comment (mbayopanda) [inline client/src/modules/fiscal/fiscal.manage.js:15]: 😆 seriously `constiables` !

**Review by mbayopanda — APPROVED** [review]
LGTM

File: client/src/modules/fiscal/fiscal.manage.js (hunk 5 above)
Comment (jniles) [reply client/src/modules/fiscal/fiscal.manage.js:15]: I was wondering where those were coming from.  Guess it was me! :P

**Pull request merged**